dataclasses_json/core.py

```diff
@@ -227,6 +227,16 @@ def _decode_dataclass(cls, kvs, infer_missing):
     return cls(**init_kwargs)
 
 
+def _decode_type(type_, value, infer_missing):
+    if _has_decoder_in_global_config(type_):
+        return _get_decoder_in_global_config(type_)(value)
+    if is_dataclass(type_) or is_dataclass(type_):
```

Comment:
If this is a refactor from the code in _decode_generic, would this line need to be
` if is_dataclass(type_) or is_dataclass(value)`

Reply:
Yes you are correct. Let me create a new PR to fix this.

Reply:
But would this fail actually, because the _decode_dataclass function looks like its expecting a dictionary in this line 
`kvs = {decode_names.get(k, k): v for k, v in kvs.items()}`